### PR TITLE
share buttons on listing gallery

### PR DESCRIPTION
**Pull request by kcvan**
## Description
Why did you write this code?
add share buttons on listing gallery
## How to Test
- [ ] Visit any listing
- [ ] Click the share button on top right
- [ ] Verify they all work

Which devices did you test on?
- [x] Chrome on Mac
- [ ] Chrome on PC
- [ ] Firefox on Mac
- [ ] Firefox on PC
- [ ] Safari iPhone
- [ ] Chrome Android

## REVIEWERS:
Check against these principles:

### High level
Does this code need to be written?
What are the alternatives?
Will this implementation become a support issue?
How much error margin does this solution have?

### Code
* Does the code follow industry standards?
JS: https://github.com/airbnb/javascript
React: https://github.com/airbnb/javascript/tree/master/react
https://github.com/vasanthk/react-bits
Documentation headers: http://usejsdoc.org/index.html

* Is there duplicated code? Can it be refactored into a shared method?
* Is the code consistent with our project?
* Are there unit tests? Do they test the states?
* Is the person refactoring another developer's code? If possible, did the original developer approve?

### Variables/Naming:
* Would the variable type led to future edge cases?
* Are the variable naming clear? Would the value contain something other than what the name describes.

### Security
* Can this be hacked or abused by the user?

## Further Work
Will this have future work?

## Learnings
Did you learn anything here you want to share with the team?

### CODE REVIEW

**Comment by kcvan** [comment]
*note

i had to use 

```
const listingLink = `https://beenest.com${location.pathname}`;
```
instead of `${BEENEST_HOST}` because facebook didnt like linking from a localhost

File: src/pages/listing/ListingGallery.tsx [hunk 1]
```diff
@@ -1,20 +1,89 @@
...
+  const [isAlertOpen, setAlert] = React.useState<boolean>(false);
+
+  const { listingPicUrl, location, photos, title } = props;
+  const listingLink = `https://beenest.com${location.pathname}`;
   return <Row className="w-100 height-60vh px-0 mx-0 position-relative">
     <ImageGrid images={[listingPicUrl, ...photos]} />
+    <Button className="position-absolute top-0 right-0 m-4" color="primary" onClick={() => toggleModal()} type="button">
+      <span className="far fa-share-square"></span>
+    </Button>
+    <Modal isOpen={isModalOpen} toggle={() => toggleModal()}>
```
Comment (woeltjen) [inline src/pages/listing/ListingGallery.tsx]: This looks good! Can we move this out to its own component? Picturing usage like:

    <ShareModal isOpen={...} toggle={...} url={listingLink} title={title} />

With this component inline, most of the lines of code in `ListingGallery` end up being about the sharing button, which is unexpected.

Comment (kcvan) [reply src/pages/listing/ListingGallery.tsx]: makes sense! will move it out to it's own component

File: src/legacy/work/ShareModal/ShareModal.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            Copied to clipboard.
+          </Alert>
+        </Row>
+      </ModalBody>
+    </Modal>
+  );
+
+  function toggleModal() {
+    toggle();
+    setAlert(!isAlertOpen);
```
Comment (jerminatorhits) [inline src/legacy/work/ShareModal/ShareModal.tsx]: When I open and close the modal, I think the alert is opening and closing as well. Maybe `setAlert(false)` for this line?

Comment (kcvan) [reply src/legacy/work/ShareModal/ShareModal.tsx]: thanks! copypasta is not always the most delicious pasta

**Pull request merged**